pysb/simulator/scipyode.py

```diff
@@ -403,36 +403,42 @@ def _test_inline(cls):
                                      extra_compile_args=extra_compile_args)
                     cls._use_inline = True
                 except (weave.build_tools.CompileError,
-                        distutils.errors.CompileError, ImportError):
-                    pass
-                except ValueError as e:
-                    if len(e.args) == 1 and \
-                                        e.args[0] == "Symbol table not found":
-                        get_logger(__name__).debug(
-                            "'ValueError: Symbol table not found' "
-                            "encountered; weave compiler is not functional")
-                    else:
+                        distutils.errors.CompileError,
+                        ImportError,
+                        ValueError) as e:
+                    if isinstance(e, ValueError) and (
+                            len(e.args) != 1 or
+                            e.args[0] != 'Symbol table not found'):
                         raise
+                    logger.warn('weave compiler appears non-functional. {}\n'
+                                'Original error: {}'.format(
+                                    _c_compiler_message(), repr(e)))
 
     @classmethod
     def _test_cython(cls):
         if not hasattr(cls, '_use_cython'):
+            logger = get_logger(__name__)
             cls._use_cython = False
             if Cython is None:
                 return
             try:
                 Cython.inline('x = 1', force=True, quiet=True)
                 cls._use_cython = True
-            except Cython.Compiler.Errors.CompileError:
-                pass
-            except ValueError as e:
-                if len(e.args) == 1 and e.args[0] == "Symbol table not found":
-                    get_logger(__name__).debug(
-                        "'ValueError: Symbol table not found' "
-                        "encountered; Cython compiler is not functional")
-                else:
+            except (Cython.Compiler.Errors.CompileError,
+                    distutils.errors.DistutilsPlatformError,
+                    ValueError) as e:
+                if isinstance(e, distutils.errors.DistutilsPlatformError) and \
+                        str(e) != 'Unable to find vcvarsall.bat':
```

Comment:
Does this error not also occur under weave.inline? Can we unify all the C compiler breakage handling into one place?